### ISSUE TITLE
Update Python dependencies

### ISSUE THREAD

**Issue by jbethune** [AKVorrat/dearmep#269]
I've tried to install DearMEP on a newish Linux machine and I got all kinds of C++ compiler errors when using Poetry.
In order to keep things simple and clean, I used the official `docker.io/library/python` image as a starting point.

There have been some changes in the world of Python packaging and various packages fail to compile when using the current `poetry.lock` file. I set the required Python version to `3.13` and I ran `poetry update` to get newer packages. I also had to install the debian package `graphviz-dev` for some header files. After that I could run `poetry install --all-extras`. I also needed to update the python package `pillow`  to version `11` from version `9.5` because of C++ compiler errors (this might be a breaking change).

Long story short: I think it would be good to update the Python dependencies in `poetry.lock` and also set the Python version to a higher number than `^3.8`. Is there something that speaks agains that? Debian bookworm is currently at python `3.11`.
